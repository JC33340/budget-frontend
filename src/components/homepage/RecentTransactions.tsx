import ItemWrapper from '../ItemWrapper';
import ItemHeading from '../ItemHeading';
import RecentTransactionsItem from './RecentTransactionsItem';
import type { transactionLogsType } from '../../pages/Transactions';

type RecentTransactionsType = {
    recentTransactions: transactionLogsType[] | undefined;
};

const RecentTransactions = ({ recentTransactions }: RecentTransactionsType) => {
    return (
        <ItemWrapper>
            <div
                className=" flex flex-col gap-y-4"
                data-testid="recentTransactions-testid"
            >
                <ItemHeading text="Recent Transactions" />
                <div className="h-60 overflow-y-scroll">
                    <div className="grid grid-cols-3 text-lg border-b-2 border-light-gray">
                        <p className="border-r-2 border-lighter-gray text-center">
                            Value
                        </p>
                        <p className="border-r-2 border-lighter-gray text-center">
                            Tag
                        </p>
                        <p className="text-center">Date</p>
                    </div>
                    {recentTransactions && recentTransactions.length > 0 ? (
                        recentTransactions?.map((item, i) => {
                            return (
                                <RecentTransactionsItem
                                    key={i}
                                    transactionLog={item}
                                    color={i % 2 === 0 ? '' : 'bg-light-gray'}
                                />
                            );
                        })
                    ) : (
                        <div className="h-full flex items-center justify-center">
                            <p>No recent transactions</p>
                        </div>
                    )}
                </div>
            </div>
        </ItemWrapper>
    );
};

export default RecentTransactions;
