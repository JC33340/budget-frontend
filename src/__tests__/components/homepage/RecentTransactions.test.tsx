import RecentTransactions from '../../../components/homepage/RecentTransactions';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

const testObj = {
    value: 100,
    tag: 'test tag',
    created_at: new Date('16 janurary 2025').toLocaleDateString(),
    notes: 'test notes',
    id: 0,
};

test('snapshot test', () => {
    render(<RecentTransactions recentTransactions={[testObj]} />);
    const comp = screen.getByTestId('recentTransactions-testid');
    expect(comp).toMatchSnapshot();
});

test('test content', () => {
    render(<RecentTransactions recentTransactions={[testObj]} />);
    const comp = screen.getByTestId('recentTransactions-testid');
    expect(comp).toHaveTextContent('test tag');
    expect(comp).toHaveTextContent('100');
    expect(comp).toHaveTextContent('16 Jan 2025');
});
