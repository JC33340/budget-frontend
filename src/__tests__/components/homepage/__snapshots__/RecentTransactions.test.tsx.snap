// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  class=" flex flex-col gap-y-4"
  data-testid="recentTransactions-testid"
>
  <p
    class="text-3xl text-blue font-medium text-center"
    data-testid="itemHeading-testid"
  >
    Recent Transactions
  </p>
  <div
    class="h-60 overflow-y-scroll"
  >
    <div
      class="grid grid-cols-3 text-lg border-b-2 border-light-gray"
    >
      <p
        class="border-r-2 border-lighter-gray text-center"
      >
        Value
      </p>
      <p
        class="border-r-2 border-lighter-gray text-center"
      >
        Tag
      </p>
      <p
        class="text-center"
      >
        Date
      </p>
    </div>
    <div
      class=" grid grid-cols-3 rounded-lg"
      data-testid="recentTransactionsItem-testid"
    >
      <p
        class="flex items-center p-2 border-r-2 border-lighter-gray"
      >
        100
      </p>
      <p
        class="flex items-center capitalize p-2 border-r-2 border-lighter-gray"
      >
        test tag
      </p>
      <p
        class="flex items-center p-2"
      >
        16 Jan 2025
      </p>
    </div>
  </div>
</div>
`;
